UnitTests/Drawing/SixelEncoderTests.cs

```diff
@@ -1,52 +1,44 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using Color = Terminal.Gui.Color;
+﻿using Color = Terminal.Gui.Color;
 
 namespace UnitTests.Drawing;
 
 public class SixelEncoderTests
 {
-
     [Fact]
     public void EncodeSixel_RedSquare12x12_ReturnsExpectedSixel ()
     {
-
-        var expected = "\u001bP" + // Start sixel sequence
-                            "0;0;0" + // Defaults for aspect ratio and grid size
-                            "q" + // Signals beginning of sixel image data
-                            "\"1;1;12;2" + // no scaling factors (1x1) and filling 12px width with 2 'sixel' height = 12 px high
-
-                            /*
-                             * Definition of the color palette
-                             */
-                            "#0;2;100;0;0" + // Red color definition in the format "#<index>;<type>;<R>;<G>;<B>" - 2 means RGB.  The values range 0 to 100
-
-                            /*
-                             * Start of the Pixel data
-                             *   We draw 6 rows at once, so end up with 2 'lines'
-                             *   Both are basically the same and terminate with dollar hyphen (except last row)
-                             *   Format is:
-                             *       #0 (selects to use color palette index 0 i.e. red)
-                             *       !12 (repeat next byte 12 times i.e. the whole length of the row)
-                             *       ~ (the byte 111111 i.e. fill completely)
-                             *       $ (return to start of line)
-                             *       - (move down to next line)
-                             */
-                            "#0!12~$-" +
-                            "#0!12~$" + // Next 6 rows of red pixels
-
-                            "\u001b\\"; // End sixel sequence
+        string expected = "\u001bP" // Start sixel sequence
+                          + "0;0;0" // Defaults for aspect ratio and grid size
+                          + "q" // Signals beginning of sixel image data
+                          + "\"1;1;12;2" // no scaling factors (1x1) and filling 12px width with 2 'sixel' height = 12 px high
```

Comment:
Just FYI, those dimensions are both in pixels, so you're filling a background size of 12x2, not 12x12. It doesn't really make any difference in this case, because you're overwriting that background with 12x12 red pixels below, but the comment is wrong.

Reply:
Thanks! seems like I should just remove these then? since they aren't doing anything and are defaults anyway (for the scaling)

Reply:
In theory you can leave off the raster attributes, but it's probably safer to include them. I know I've encountered terminal emulators in the past that couldn't handle sixel images if the size wasn't set (although that was several years ago).

Also if you do remove the, then you must set the first two parameters to 9 and 1, i.e. the sequence should start `\u001bP9;1q`. Because the default aspect ratio is actually 2:1 and the default background fill will cover the maximum extent of the page. Setting P1 to 9 gives you a 1:1 aspect ratio, and setting P2 to 1 will disable the background fill so you don't need to worry about the size.

Reply:
>  those dimensions are both in pixels, so you're filling a background size of 12x2, not 12x12. 

Thanks I have updated in [4571978](https://github.com/gui-cs/Terminal.Gui/pull/3734/commits/4571978a9b84f02886bfee539d07a020454e24d6).

@j4james do you know if there is a reference source for sixel online? I have been mostly working off other MIT open source libraries ([node-sixel]( https://github.com/jerch/node-sixel/) and [libsixel](https://github.com/saitoha/libsixel)) and the [Wikipedia article](https://en.wikipedia.org/wiki/Sixel) and [Iterm 2 docs](https://iterm2.com/documentation-images.html)

Reply:
I think the best official documentation for Sixel is the VT340 Graphics Programming manual. You should find several web sites hosting copies of that if you search for `EK-VT3XX-GP`.

But if you're interested in edge cases not covered in the official docs, the best source of information is the [vt340test](https://github.com/hackerb9/vt340test) repository. That has numerous test cases with screenshots of their output from a real VT340 terminal.

Reply:
And if you're looking for examples of other libraries which do Sixel encoding, these are a few that I'm aware of:

* https://gitlab.com/AutumnMeowMeow/jexer
* https://github.com/dankamongmen/notcurses
* https://github.com/hpjansson/chafa